src/TempleLineOfCreditV2.sol

```diff
@@ -0,0 +1,368 @@
+        uint256 _oudMinCollateralizationRatio,
+
+        address _oudRedeemer
+
+    ) {
+
+        templeToken = IERC20(_templeToken);
+        templePrice = _templePrice;
+
+        dai.tokenAddress =_daiToken;
+        dai.interestRateModel = _daiInterestRateModel;
+        dai.tokenPrice = _daiPrice;
+        dai.minCollateralizationRatio = _daiMinCollateralizationRatio; 
+
+        oud.tokenAddress = _oudToken;
+        oud.interestRateModel = _oudInterestRateModel;
+        oud.tokenPrice = _oudPrice;
+        oud.minCollateralizationRatio = _oudMinCollateralizationRatio;
+
+        oudRedeemer = IOudRedeemer(_oudRedeemer);
+    }
+
+    function addOperator(address _address) external override onlyOwner {
+        _addOperator(_address);
+    }
+
+    function removeOperator(address _address) external override onlyOwner {
+        _removeOperator(_address);
+    }
+
+    /**
+     * @notice Set the borrow interest rate for debt token
+     * @param debtToken address of the debt token
+     * @param interestRateModel the interest rate model to use
+     */
+    function setInterestRateModel(address debtToken, address interestRateModel) external onlyOperators{
+
+        if (debtToken == dai.tokenAddress) {
+            _accrueInterest(dai);
+            dai.interestRateModel = interestRateModel;
+        } else if (debtToken == oud.tokenAddress) {
+            _accrueInterest(oud);
+            oud.interestRateModel = interestRateModel;
+        } else {
+            revert Unsupported(debtToken);
+        }
+    }
+
+
+    function _accrueInterest(ReserveToken storage reserve) internal {
+
+        uint256 totalBorrow = reserve.totalBorrow;
+        uint256 totalReserve = reserve.totalReserve;
+        uint256 totalShares = reserve.totalShares;
+
+        // If no borrow just reset interest updated at.
+        if (totalShares == 0) {
+
+            reserve.interestRateLastUpdatedAt = block.timestamp;
+
+        } else {
+
+            uint256 timeElapsed = block.timestamp - reserve.interestRateLastUpdatedAt;
+            uint256 newInterestRate =  IInterestRateModel(reserve.interestRateModel).getBorrowRate(totalBorrow, totalReserve);
+
+            uint256 interestEarned = totalBorrow.continuouslyCompounded(timeElapsed, newInterestRate) - totalBorrow;
+
+            totalBorrow += interestEarned;
+            totalReserve += interestEarned; // Reserve should also increase to properly account for utilization rate
+
+            reserve.totalBorrow = totalBorrow;
+            reserve.totalReserve = totalReserve;
+            reserve.interestRateLastUpdatedAt = block.timestamp;
+
+            emit InterestRateUpdate(reserve.tokenAddress, newInterestRate);
+        }
+    }
+
+    /**
+     * @notice Allows operator to deposit debt tokens
+     * @param account account to take debtToken from 
+     * @param amount is the amount to deposit
+     */
+    function depositDAIReserve(address account, uint256 amount) external onlyOperators{
```

Comment:
This should just be pulling from the TreasuryReservesVault on demand (and depositing any reserves). There shouldn't be any explicit dai balance in this contract now

Reply:
to make sure we have the accounting process right. This method will pull dai put it into treasury vault and track total reserves